src/viur/core/skeleton.py

```diff
@@ -1968,34 +1968,60 @@ def canCall(self) -> bool:
         return user is not None and "root" in user["access"]
 
     def dataSkel(self):
-        modules = ["*"] + listKnownSkeletons()
-        modules.sort()
         skel = BaseSkeleton().clone()
-        skel.module = SelectBone(descr="Module", values={x: translate(x) for x in modules}, required=True)
+
+        skel.skelname = SelectBone(
+            descr="Kind",
+            values={"*": "*"} | {name: translate(name) for name in sorted(listKnownSkeletons())},
+        )
+
+        skel.specific_key = StringBone(
+            descr="Key",
+            params={
+                "tooltip": "Optional: Key of one entry to refresh, ignores the kind selected above."
+            }
+        )
+
         return skel
 
-    def execute(self, module, *args, **kwargs):
+    def execute(self, skelname, specific_key, *args, **kwargs):
         usr = current.user.get()
         if not usr:
             logging.warning("Don't know who to inform after rebuilding finished")
             notify = None
         else:
             notify = usr["name"]
 
-        if module == "*":
-            for module in listKnownSkeletons():
-                logging.info("Rebuilding search index for module %r", module)
-                self._run(module, notify)
+        if skelname == "*":
+            if specific_key:
+                raise errors.BadRequest(f"Cannot run on all kinds with {specific_key=}")
```

Comment:
Why we raise an error here in the tooltip above is written that the kind is ignored.

Reply:
Because skelname == "*" defines "all kinds"... an iteration over all available kinds. Therefore, it should not be usable with a key. I also made it this way because I was lazy... there could be more effort done into rewriting this task, but we have so many construction sites currently...

Reply:
Why don't we just ignore the kind as soon as a `specific_key` is provided?

With `params={"logic.visibleIf": "bool(specific_key)"},` we could even display this in the admin.

Reply:
It is implemented as such. The "*" must be explicitly selected, I removed required from skelname within this PR, so that either a skelname or a key can be provided.

Reply:
well okay.